lib/phonelib/phone_analyzer.rb

```diff
@@ -102,11 +102,11 @@ def parse_single_country(e164, data)
     # ==== Attributes
     #
     # * +phone+ - phone number for parsing
-    def detect_and_parse(phone)
+    def detect_and_parse(phone, country = nil)
```

Comment:
Don't think `country` needs to be optional. At least, there's only one invocation and it explicitly provides that argument.